sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1687,6 +1687,15 @@ object SQLConf {
       .booleanConf
       .createWithDefault(false)
 
+  val LEGACY_PASS_PARTITION_BY_AS_OPTIONS =
+    buildConf("spark.sql.legacy.sources.write.passPartitionByAsOptions")
+      .internal()
+      .doc("Whether to pass the partitionBy columns as options in DataFrameWriter." +
+        " Data source V1 now silently drops partitionBy columns for non-file-format sources;" +
+        " turning the flag on provides a way for these sources to see these partitionBy columns.")
+      .booleanConf
+      .createWithDefault(true)
```

Comment:
shouldn't the default be `false` to be backward compatible?

Reply:
I went back and forth on what to recommend here.  I think we need to balance the fact that this is a behavior change with the confusion that a user will experience when options they specify are silently dropped.

I recommended going with `true` because: a) I don't know any V1 sources that validate options where this change would break an existing program and b) the only time behavior changes is when someone is specifying a (silently dropped) partitionBy clause.

Thoughts?